tests/integration/workflows/test_arxiv_merge.py

```diff
@@ -0,0 +1,339 @@
+# or submit itself to any jurisdiction.
+
+"""Tests for merge and resolve merge conflicts."""
+
+from __future__ import absolute_import, division, print_function
+
+import json
+import mock
+import pytest
+
+from invenio_workflows import (
+    ObjectStatus,
+    WorkflowEngine,
+    start,
+)
+
+from inspirehep.modules.records.api import InspireRecord
+
+
+@pytest.fixture
+def enable_merge_on_update(workflow_app):
+    with mock.patch.dict(workflow_app.config, {'FEATURE_FLAG_ENABLE_MERGE_ON_UPDATE': True}):
+        yield
+
+
+def test_merge_with_disabled_merge_on_update_feature_flag(workflow_app, record_to_merge):
+    record_update = {
+        '$schema': 'http://localhost:5000/schemas/records/hep.json',
+        '_collections': ['Literature'],
+        'document_type': ['article'],
+        'titles': [
+            {'title': 'Jessica Jones'},
+            {'title': 'Luke Cage'},
+            {'title': 'Frank Castle'},
+        ],
+        'dois': [
+            {
+                'value': '10.1007/978-3-319-15001-7'
+            }
+        ],
+    }
+
+    eng_uuid = start('article', [record_update])
+
+    eng = WorkflowEngine.from_uuid(eng_uuid)
+    obj = eng.objects[0]
+
+    assert obj.extra_data.get('callback_url') is None
+    assert obj.extra_data.get('conflicts') is None
+    assert obj.extra_data.get('merged') is None
+
+    assert obj.extra_data.get('skipped-store-record') is True
+    assert obj.extra_data.get('skipped-robot-upload') is True
+
+    record = InspireRecord.get_record(record_to_merge.id)
+
+    expected_titles = [
+        {'title': 'Alias Investigations'},
+    ]
+
+    assert expected_titles == record.get('titles')
+
+
+def test_merge_without_conflicts(workflow_app, enable_merge_on_update, record_to_merge):
+    record_update = {
+        '$schema': 'http://localhost:5000/schemas/records/hep.json',
+        '_collections': ['Literature'],
+        'document_type': ['article'],
+        'titles': [
+            {'title': 'Jessica Jones'},
+            {'title': 'Luke Cage'},
+            {'title': 'Frank Castle'},
+        ],
+        'dois': [
+            {
+                'value': '10.1007/978-3-319-15001-7'
+            }
+        ],
+    }
+
+    eng_uuid = start('article', [record_update])
+
+    eng = WorkflowEngine.from_uuid(eng_uuid)
+    obj = eng.objects[0]
+
+    assert obj.status == ObjectStatus.COMPLETED
+    assert obj.extra_data.get('callback_url') is None
+    assert obj.extra_data.get('conflicts') is None
+
+    assert obj.extra_data.get('approved') is True
+    assert obj.extra_data.get('is-update') is True
+    assert obj.extra_data.get('merged') is True
+
+    record = InspireRecord.get_record(record_to_merge.id)
+
+    expected_titles = [
+        {'title': 'Alias Investigations'},
+        {'title': 'Jessica Jones'},
+        {'title': 'Luke Cage'},
+        {'title': 'Frank Castle'},
+    ]
+
+    assert expected_titles == record.get('titles')
```

Comment:
I would remove this assertion: if, for any reason, the `inspire-json-merger` library changes its internal merging rules, this test is going to fail, while it should still pass